frontend/catalyst/ag_primitives.py

```diff
@@ -134,9 +319,95 @@ def converted_call(fn, *args, **kwargs):
 
             @functools.wraps(fn.func)
             def qnode_call_wrapper():
-                return tf_converted_call(fn.func, *args, **kwargs)
+                return tf_converted_call(fn.func, args, kwargs, caller_fn_scope, options)
 
             new_qnode = qml.QNode(qnode_call_wrapper, device=fn.device, diff_method=fn.diff_method)
             return new_qnode()
 
-        return tf_converted_call(fn, *args, **kwargs)
+        return tf_converted_call(fn, args, kwargs, caller_fn_scope, options)
+
+
+class CRange:
```

Comment:
is there ever a situation for a user to use this directly?

Reply:
I don't think so? A catalyst for loop doesn't use a range object, a Python loop without autograph would still fail if they used a range with dynamic bounds (whether they use builtin `range` or `CRange`), and a Python loop with autograph already converts the builtin `range` to our `CRange`.

Reply:
Yep, makes sense!